### DISCUSSION TITLE
Convert to onnx after script removal?

### DISCUSSION THREAD

**Discussion by christian-holland** [open-mmlab/mmclassification#1540] [Q&A] [answered]
I want to convert a trained model to onnx. It looks like previously there was a pytorch2onnx.py script, but it looks like this doesn't exist anymore. I assume this change was made, so that the [mmdeploy](https://github.com/open-mmlab/mmdeploy/blob/main/docs/en/02-how-to-run/convert_model.md) repo is used instead.

However it is not entirely clear to me from the docs there, how it would be used in that way. Trying to follow the steps as I understood them from the mmdeploy docs, I tried:

- Build the docker image like described [here](https://github.com/open-mmlab/mmdeploy/blob/main/docs/en/01-how-to-build/build_from_docker.md) and installed the mmpretrain repo from source.
- Run the conversion like as described [here](https://github.com/open-mmlab/mmdeploy/blob/main/docs/en/01-how-to-build/build_from_docker.md).

In particular I tried:
```bash
python tools/torch2onnx.py \
    configs/mmcls/classification_onnxruntime_dynamic.py \
    /data/mmpretrain/work_dirs/swinv2-tiny-w16_16xb64_in1k-256px/swinv2-tiny-w16_16xb64_in1k-256px.py \
    /data/mmpretrain/work_dirs/swinv2-tiny-w16_16xb64_in1k-256px/epoch_300.pth \
    /data/FST/221130_person_crops/val/yellow_person/img_2022_09_14_10_31_45t578460.jpg \
    --work-dir /data/mmpretrain/work_dirs/swinv2-tiny-w16_16xb64_in1k-256px
```

I get this error `ImportError: mmcls has not been installed. Import mmcls failed.`  This is just the old name of this repo? However, this made me question if I am trying to do the correct thing in the first place here for what I am trying to achieve.

Can you give me some guidance what would be intended way if I want to convert a trained model to onnx with this repo?

**Comment by RunningLeon** [comment, marked as answer]
@christian-holland hi，could use mmdeploy after this PR https://github.com/open-mmlab/mmdeploy/pull/2003

**Reply by christian-holland** [reply]
Thanks. The Dockerfile uses `master` as default where these changes are not yet present. Building from the `main` branch resolves the issue.

**Reply by RunningLeon** [reply]
Good to hear that. If you mean the dockerfile in mmdeploy, pls. use the latest `main` branch, which has already updated to use `main` branch as per [here](https://github.com/open-mmlab/mmdeploy/blob/840adcfb4319c483a8d0f548a0c491b361df8214/docker/CPU/Dockerfile#L88) and [here](https://github.com/open-mmlab/mmdeploy/blob/840adcfb4319c483a8d0f548a0c491b361df8214/docker/GPU/Dockerfile#L68).
